README.md

```diff
@@ -48,96 +48,120 @@ your Cloud SQL instance.
-- The [Cloud SQL Admin API][admin-api] to be enabled within your Google Cloud
-Project. By default, the API will be called in the project associated with
-the IAM principal.
+#### Postgres
 
-[admin-api]: https://console.cloud.google.com/apis/api/sqladmin.googleapis.com
-[client-role]: https://cloud.google.com/sql/docs/mysql/roles-and-permissions
+To use `database/sql`, use `pgxv4.RegisterDriver` with any necessary Dialer
+configuration. Note: the connection string must use the keyword/value format
+with host set to the instance connection name.
 
-### Credentials
+``` go
+import (
+    "database/sql"
 
-This repo uses the [Application Default Credentials (ADC)][adc] strategy for
-resolving credentials. Please see [these instructions for how to set your ADC][set-adc]
-(Google Cloud Application vs Local Development, IAM user vs service account credentials),
-or consult the [golang.org/x/oauth2/google][google-auth] documentation.
+    "cloud.google.com/go/cloudsqlconn"
+    "cloud.google.com/go/cloudsqlconn/postgres/pgxv4"
+)
 
-To explicitly set a specific source for the Credentials, see [Using
-Options](#using-options) below.
+func connect() {
+    cleanup, err := pgxv4.RegisterDriver("cloudsql-postgres", cloudsqlconn.WithIAMAuthN())
+    if err != nil {
+        // ... handle error
+    }
+    defer cleanup()
 
-[adc]: https://cloud.google.com/docs/authentication#adc
-[set-adc]: https://cloud.google.com/docs/authentication/provide-credentials-adc
-[google-auth]: https://pkg.go.dev/golang.org/x/oauth2/google#hdr-Credentials
+    db, err := sql.Open(
+        "cloudsql-postgres",
+        "host=project:region:instance user=myuser password=mypass dbname=mydb sslmode=disable",
+	)
+    // ... etc
+}
+```
 
-### Postgres
+#### MySQL
 
-To use the dialer with [pgx](https://github.com/jackc/pgx), use
-[pgxpool](https://pkg.go.dev/github.com/jackc/pgx/v4/pgxpool) by configuring
-a [Config.DialFunc][dial-func] like so:
+To use `database/sql`, use `mysql.RegisterDriver` with any necessary Dialer
+configuration.
 
-``` go
+```go
 import (
-	"context"
-	"net"
+    "database/sql"
 
-	"cloud.google.com/go/cloudsqlconn"
-	"github.com/jackc/pgx/v4/pgxpool"
+    "cloud.google.com/go/cloudsqlconn"
+    "cloud.google.com/go/cloudsqlconn/mysql/mysql"
 )
 
 func connect() {
-	// Configure the driver to connect to the database
-	dsn := "user=myuser password=mypass dbname=mydb sslmode=disable"
-	config, err := pgxpool.ParseConfig(dsn)
-	if err != nil {
-		/* handle error */
-	}
+    cleanup, err := mysql.RegisterDriver("cloudsql-mysql", cloudsqlconn.WithCredentialsFile("key.json"))
+    if err != nil {
+        // ... handle error
+    }
+    defer cleanup()
 
-	// Create a new dialer with any options
-	d, err := cloudsqlconn.NewDialer(context.Background())
-	if err != nil {
-		/* handle error */
-	}
-	defer d.Close()
+    db, err := sql.Open(
+        "cloudsql-mysql",
+        "myuser:mypass@cloudsql-mysql(my-project:us-central1:my-instance)/mydb",
```

Comment:
I would try and remain consistent with variable naming across database engines for your instance connection name. 
```suggestion
        "myuser:mypass@cloudsql-mysql(project:instance:instance)/mydb",
```